### PR TITLE
Extract HTTP client experimental metrics to a separate class

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**

### CODE REVIEW

File: javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/internal/CommonConfig.java [hunk 1]
```diff
@@ -43,6 +44,8 @@ public static CommonConfig get() {
         config.getList("otel.instrumentation.http.capture-headers.server.response", emptyList());
     statementSanitizationEnabled =
         config.getBoolean("otel.instrumentation.common.db-statement-sanitizer.enabled", true);
+    emitExperimentalHttpClientMetrics =
+        config.getBoolean("otel.instrumentation.http.client.emit-experimental-metrics", false);
```
Comment (mateuszrzeszutek) [inline javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/internal/CommonConfig.java:48]: I've also considered `...experimental-metrics.enabled`, `...capture-experimental-metrics` but honestly I wasn't really sure which one of them is better (or less bad)

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientExperimentalMetrics.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.instrumentation.api.instrumenter.OperationListener;
+import io.opentelemetry.instrumentation.api.instrumenter.OperationMetrics;
+import java.util.logging.Logger;
+
+/**
+ * {@link OperationListener} which keeps track of <a
+ * href="https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/http-metrics.md#http-client">non-stable
```
Comment (breedx-splk) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientExperimentalMetrics.java:24]:
```suggestion
 * href="https://github.com/open-telemetry/semantic-conventions/blob/main/specification/metrics/semantic_conventions/http-metrics.md#http-client">experimental
```

just to keep the wording consistent.

File: instrumentation/java-http-client/library/src/main/java/io/opentelemetry/instrumentation/httpclient/JavaHttpClientTelemetryBuilder.java [hunk 3]
```diff
@@ -65,10 +66,27 @@ public JavaHttpClientTelemetryBuilder setCapturedResponseHeaders(List<String> re
...
+
   public JavaHttpClientTelemetry build() {
     Instrumenter<HttpRequest, HttpResponse<?>> instrumenter =
         JavaHttpClientInstrumenterFactory.createInstrumenter(
-            openTelemetry, capturedRequestHeaders, capturedResponseHeaders, additionalExtractors);
+            openTelemetry,
+            capturedRequestHeaders,
+            capturedResponseHeaders,
+            additionalExtractors,
+            emitExperimentalHttpClientMetrics);
```
Comment (breedx-splk) [inline instrumentation/java-http-client/library/src/main/java/io/opentelemetry/instrumentation/httpclient/JavaHttpClientTelemetryBuilder.java:89]: Nearly time for a factory factory. 🤣

File: javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/internal/CommonConfig.java (hunk 1 above)
Comment (breedx-splk) [reply javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/internal/CommonConfig.java:48]: Seems fine this way to me.

**Review by breedx-splk — APPROVED** [review]
Looks like a perfectly reasonable approach. Thanks for being so thorough!

File: instrumentation/java-http-client/library/src/main/java/io/opentelemetry/instrumentation/httpclient/JavaHttpClientTelemetryBuilder.java (hunk 3 above)
Comment (mateuszrzeszutek) [reply instrumentation/java-http-client/library/src/main/java/io/opentelemetry/instrumentation/httpclient/JavaHttpClientTelemetryBuilder.java:89]: factory^2

**Comment by mateuszrzeszutek** [comment]
@trask do you want to take a look at it before merging it?

**Pull request merged**